src/core/qgsvectorfilewriter.cpp

```diff
@@ -1551,6 +1565,10 @@ bool QgsVectorFileWriter::driverMetadata( const QString& driverName, QgsVectorFi
   return false;
 }
 
+OGRwkbGeometryType QgsVectorFileWriter::ogrTypeFromWkbType( QgsWKBTypes::Type type )
+{
+  return static_cast<OGRwkbGeometryType>( type );
```

Comment:
Measure values aren't supported by gdal - they are converted by some providers to z values (See http://www.gdal.org/drv_shapefile.html). We could either do the same or drop them. CircularStringZ/... curved geometries with z are supported in gdal >= 2.0